### PR TITLE
More for LOX

### PR DESCRIPTION
**Pull request by JohannesMeierSE**
- fixed some bugs
- more test cases
- validations: support for removing validations rules of removed types
- classes:
  - support methods/functions in classes
  - UniqueMethodValidation
  - implemented AnyClassType
- refactorings
- unified configurations for OX and LOX

### CODE REVIEW

File: packages/typir/src/kinds/class-kind.ts [hunk 1]
```diff
@@ -550,3 +604,232 @@ export class UniqueClassValidation implements ValidationRuleWithBeforeAfter {
...
+        }
+        return [];
+    }
+
+    /**
+     * Calculates a key for a method which encodes its unique properties, i.e. duplicate methods have the same key.
+     * Additionally, the class of the method needs to be represented in the key as well.
+     * This key is used to identify duplicated methods.
+     * Override this method to change the properties which make a method unique.
+     * @param clas the current class type
```
Comment (insafuhrmann) [inline packages/typir/src/kinds/class-kind.ts:657]: Why clas with only one s?

File: packages/typir/src/kinds/class-kind.ts [hunk 2]
```diff
@@ -550,3 +604,232 @@ export class UniqueClassValidation implements ValidationRuleWithBeforeAfter {
...
+            }
+        }
+
+        this.foundDeclarations.clear();
+        return result;
+    }
+}
+
+
+// TODO for the review: which name is better? AnyClassType vs TopClassType?
```
Comment (insafuhrmann) [inline packages/typir/src/kinds/class-kind.ts:686]: I have no strong personal preference. TopClassType leaves no doubt what this is for but AnyClassType is nicer regarding natural language imho. However, maybe we should just decide based on consistency.

**Review by insafuhrmann — APPROVED** [review]
Thank you very much @JohannesMeierSE, this looks like a helpful set of additions. I have only a few detail remarks.

File: examples/lox/test/lox-type-checking.test.ts [hunk 3]
```diff
@@ -183,6 +239,57 @@ describe('Explicitly test type checking for LOX', () => {
...
+        class MyClass1 {
+            method1(input: number): number {
+                return 123;
+            }
+            method1(another: number): boolean {
+                return true;
+            }
+        }
+    `, 2)); // both methods need to be marked
+
```
[the remark this answers is not in the document]
Comment (JohannesMeierSE) [reply examples/lox/test/lox-type-checking.test.ts:292]: Thanks!

File: packages/typir/src/kinds/class-kind.ts (hunk 2 above)
Comment (JohannesMeierSE) [reply packages/typir/src/kinds/class-kind.ts:686]: Thanks for your argumentation! I agree: To be consistent with `TopType`, I renamed `AnyClassType` to `TopClassType`.

File: examples/lox/test/lox-type-checking.test.ts [hunk 4]
```diff
@@ -210,6 +317,81 @@ describe('Test internal validation of Typir for cycles in the class inheritance
     });
 });
 
+describe('LOX', () => {
+    test.fails('complete with difficult order of classes', async () => await validate(`
```
[the remark this answers is not in the document]
Comment (JohannesMeierSE) [reply examples/lox/test/lox-type-checking.test.ts:321]: You are completely right, thank you very much! I replaced `test.fails` by `test.todo` and investigated/fixed the other occurrances of `test.fails` as well.

**Comment by JohannesMeierSE** [comment]
Thanks @insafuhrmann for your helpful review! I just pushed my fixes.

File: packages/typir/src/kinds/class-kind.ts (hunk 1 above)
Comment (insafuhrmann) [reply packages/typir/src/kinds/class-kind.ts:657]: Ok for me, I would have used classT for class type (or something else that does not look like a typo), but that is not really important

**Pull request merged**